core/src/world/place/location/geographical/canyon.rs

```diff
@@ -0,0 +1,179 @@
+use crate::{
+    utils::pluralize,
+    world::{Demographics, Place},
+};
+use rand::prelude::*;
+
+pub fn generate(place: &mut Place, rng: &mut impl Rng, _demographics: &Demographics) {
+    place.name.replace_with(|_| name(rng));
+}
+
+fn name(rng: &mut impl Rng) -> String {
+    match rng.gen_range(0..=5) {
+        0 => format!("{} {}", thing(rng), canyon_synonym(rng)),
+        1 => format!("The {} {}", placement(rng), canyon_synonym(rng)),
+        2 => format!("{} {}", cardinal_direction(rng), canyon_synonym(rng)),
+        3 => format!("{} {}", adjective(rng), canyon_synonym(rng)),
+        4 => format!("{} {} {}", adjective(rng), thing(rng), canyon_synonym(rng)),
+        5 => {
+            let (profession, s) = pluralize(profession(rng));
+            format!("{}{} {}", profession, s, canyon_synonym(rng))
+        }
+        _ => unreachable!(),
+    }
+}
+
+fn thing(rng: &mut impl Rng) -> &'static str {
+    match rng.gen_range(0..=9) {
+        0 => other_animal(rng),
+        1..=4 => canyon_animal(rng),
+        5 => enemy(rng),
+        6 => food(rng),
+        7 => profession(rng),
+        8 => symbol(rng),
+        9 => gem(rng),
+        _ => unreachable!(),
+    }
+}
+
+fn adjective(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const ADJECTIVES: &[&str] = &[
+        "Angry", "Awkward", "Bloated", "Driven", "Elegant", "Engorged",
+        "Enchanted", "Gold", "Green", "Grey", "Grouchy", "Hallowed", "Happy",
+        "Hidden", "Hungry", "Jovial", "Lone", "Lost", "Lucky", "Merry", "Moody",
+        "Morose", "Quiet", "Red", "Sickly", "Silent", "Silver", "Slim",
+        "Slender", "Solemn", "Starved", "Stoic", "Thirsty", "Wasted", "Wild",
+        "White", "Yellow"
+    ];
+    ADJECTIVES[rng.gen_range(0..ADJECTIVES.len())]
+}
+
+fn canyon_synonym(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const CANYON_SYNONYMS: &[&str] = &[
+        "Canyon", "Ravine", "Gorge", "Crevice", "Gorge", "Gap", "Abyss",
```

Comment:
I presume Gorge is in there twice to give it heavier weight? Not a bad idea, although I'd actually dial it up some more, and maybe boost Canyon +2 and Ravine +1 as well.

Reply:
Actually I think that's in error, but yes I'll absolutely do that.

Reply:
Good error. There's also [`rand::distributions::WeightedIndex`](https://docs.rs/rand/0.8.5/rand/distributions/struct.WeightedIndex.html) available, but it seems like overkill for this use case (and it allocates, for all we care about that in this not-particularly-hot code path).